.github/workflows/build.yml

```diff
@@ -577,14 +644,17 @@ jobs:
       - uses: actions/setup-java@v4
         with:
           distribution: temurin
-          # here we intentionally use an older version so that we also verify Java 17 compiles to it
```

Comment:
Is this comment actually true?
I know Java compiled on an old version should work with newer versions: This is not the first time I run into the issue with a old version trying something compiled with a new version.
The logs print this:
```

Reply:
cc @jimexist to chime in.

Reply:
it was not a typo.

that `UnsupportedClassVersionError` likely from kotlin which isn't configured with the same version as the java lib project?

Reply:
I created a ticket: [THRIFT-5879: java and kotlin cross tests fail in the GitHub action](https://issues.apache.org/jira/browse/THRIFT-5879)